src/operator/contrib/fft-inl.h

```diff
@@ -0,0 +1,306 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file fft-inl.h
+ * \brief
+ * \author Chen Zhu
+*/
+#ifndef MXNET_OPERATOR_CONTRIB_FFT_INL_H_
+#define MXNET_OPERATOR_CONTRIB_FFT_INL_H_
+#ifdef MSHADOW_USE_CUDNN
+#include <cufft.h>
+#endif
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <map>
+#include <vector>
+#include <string>
+#include <utility>
+#include <iostream>
+#include "../operator_common.h"
+#include "../mshadow_op.h"
+
+namespace mxnet {
+namespace op {
+namespace fft {
+enum fftOpInputs {kData};
+enum fftOpOutputs {kOutComplex};  // seperate the image and real parts at the moment
+enum fftOpResource {kTempSpace};  // might be requiered as we need to pad the real matrices
+}
+
+struct FFTParam : public dmlc::Parameter<FFTParam> {
+  int compute_size;  // the maximum size of sub-batch to be forwarded through FFT in one time
```

Comment:
In CuFFT: cufftPlanMany(), they defined this. So batches(=number of fft/compute size) can be executed in parallel.